## 1. Examining the HyperKvasir Dataset
### 1.1. Image labels

In [23]:
import pandas as pd

df1 = pd.read_csv('file-names/image-labels.csv')
print(df1.shape)
df1.drop(df1.loc[df1['Organ'] != 'Upper GI'].index, inplace=True)
print(df1.shape)
df1.drop(df1.loc[df1['Classification'] != 'anatomical-landmarks'].index, inplace=True)
df1.reset_index(drop=True, inplace=True)
print(df1.shape)
print(df1['Finding'].value_counts())
df1.to_csv('file-names/filtered-names/image-labels.csv')

10662
3452
2695
pylorus              999
z-line               932
retroflex-stomach    764
Name: Finding, dtype: int64


| # of Labeled Images | # of Upper GI Labels | # of Anatomical Landmark Labels | # of Z-line Labels | # of Retroflex Stomach Labels | # of Pylorus Labels |
| - | - | - | - | - | - |
| 10662 | 3452 | 2695 | 932 | 764 | 999 |

### 1.2. Video Labels

In [40]:
lis1 = []
df2 = pd.read_csv('file-names/video-labels.csv')
print(df2.shape)
df2.drop(df2.loc[df2['Organ'] != 'Upper GI'].index, inplace=True)
df2.reset_index(drop=True, inplace=True)
print(df2.shape)
for i in range(df2.shape[0]):
    if (df2['Classification 1'].iloc[i] != 'Anatomical landmarks') \
        & (df2['Classification 2'].iloc[i] != 'Anatomical landmarks'):
            lis1.append(i)
df2.drop(df2.index[lis1], inplace=True)
df2.reset_index(drop=True, inplace=True)
print(df2.shape)


(374, 7)
(60, 7)
(12, 7)


In [39]:
for i in range(df2.shape[0]):
    if (df2['Classification 1'].iloc[i] == 'Anatomical landmarks') \
        & (df2['Classification 2'].iloc[i] == 'Anatomical landmarks'):
            lis1.append(i)
count1 = df2['Finding 1'].value_counts() 
count2= df2['Finding 2'].value_counts()
count3 = count1.add(count2, fill_value=0)
print(count3)
df2.to_csv('file-names/filtered-names/video-labels.csv')

Placement of duodenal tube               1.0
Z-line                                   5.0
distal oesophagus,                       1.0
duodenal diverticula                     1.0
duodenum                                 1.0
hematine, mix of old blod and liquide    1.0
polyps                                   1.0
pylorus                                  6.0
reduced view                             1.0
retroflex stomach                        2.0
dtype: float64


| # of Labeled Videos | # of Upper GI Labels | # of Anatomical Landmark Labels | # of Z-line Labels | # of Retroflex Stomach Labels | # of Pyrolus Labels |
| - | - | - | - | - | - |
| 374 | 60 | 12 | 5 | 2 | 6 | 